Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/DiagonalVelocityDampingForceField.h

```diff
@@ -45,6 +45,9 @@ class DiagonalVelocityDampingForceField : public core::behavior::ForceField<Data
     typedef core::objectmodel::Data<VecDeriv> DataVecDeriv;
 
     /// air drag coefficient.
+    Data< VecDeriv > d_dampingCoefficients;
+
+    SOFA_ATTRIBUTE_DEPRECATED("v23.12", "v24.06", "This Data is now replaced by d_dampingCoefficients")
     Data< VecDeriv > dampingCoefficients;
```

Comment:
```suggestion
    union
    {
        Data< VecDeriv > d_dampingCoefficients;
        SOFA_ATTRIBUTE_DEPRECATED("v23.12", "v24.06", "This Data is now replaced by d_dampingCoefficients")
        Data< VecDeriv > dampingCoefficients;
    }
```

I think I have found a better way to handle such deprecation. See a simple example https://godbolt.org/z/d3xPEGoh7. This way, if an external code uses `dampingCoefficients`, it will behaves exactly as if it uses `d_dampingCoefficients`, except that a warning will be triggered.

Reply:
wow, well done !! 👍